container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/user_services_functions/start_user_services.go

```diff
@@ -129,11 +134,20 @@ func StartUserServices(
 		return successfulServicesPool, failedServicesPool, nil
 	}
 
+	for _, serviceRegistrationToStart := range serviceRegistrationsToStart {
+		if serviceRegistrationToStart.GetStatus() == service.ServiceStatus_Stopped {
+			// If the first service to start is stopped, we know that the other ones are too because
+			// of a check done at the service network layer.
+			// Restarting stopped services is a much lighter operation so we branch out to a simpler function
+			return restartUserServices(ctx, enclaveUuid, serviceRegistrationsToStart, dockerManager)
```

Comment:
I could not find a clean way yet on using a single method to handle both first start and restart use cases.  There is added complexity around parallelization and create+start container versus start container.  I can revisit later if needed.